pkg/standalone-utils/contracts/ProtocolFeeSplitter.sol

```diff
@@ -0,0 +1,152 @@
+        _require(feePercentage <= _MAX_REVENUE_SHARING_FEE_PERCENTAGE, Errors.SPLITTER_FEE_PERCENTAGE_TOO_HIGH);
+        defaultRevenueSharingFeePercentage = feePercentage;
+        emit DefaultRevenueSharingFeePercentageChanged(feePercentage);
+    }
+
+    function setPoolBeneficiary(bytes32 poolId, address newBeneficiary) external override {
+        _require(msg.sender == Pool(VaultHelpers.toPoolAddress(poolId)).getOwner(), Errors.SENDER_NOT_ALLOWED);
+        poolSettings[poolId].beneficiary = newBeneficiary;
+        emit PoolBeneficiaryChanged(poolId, newBeneficiary);
+    }
+
+    function collectFees(bytes32 poolId) external override {
+        address pool = VaultHelpers.toPoolAddress(poolId);
+        IERC20 bpt = IERC20(pool);
+
+        IProtocolFeesCollector protocolFeesCollectorMemory = protocolFeesCollector;
+        uint256 feeCollectorBptBalance = bpt.balanceOf(address(protocolFeesCollectorMemory));
+        if (feeCollectorBptBalance == 0) {
+            _revert(Errors.NO_BPT_FEES_COLLECTED);
+        }
+
+        // withdrawCollectedFees requires array of tokens, and we are only transfering BPT token
+        IERC20[] memory tokens = new IERC20[](1);
+        tokens[0] = bpt;
+
+        address beneficiary = poolSettings[poolId].beneficiary;
+
+        if (beneficiary == address(0)) {
+            uint256[] memory amounts = new uint256[](1);
+            amounts[0] = feeCollectorBptBalance;
+            protocolFeesCollectorMemory.withdrawCollectedFees(tokens, amounts, treasury);
+            emit FeesCollected(poolId, beneficiary, 0, treasury, feeCollectorBptBalance);
+        } else {
+            uint256[] memory amounts = new uint256[](1);
+
+            uint256 poolFeeOverride = poolSettings[poolId].revenueSharePercentageOverride;
+            uint256 feePercentage = poolFeeOverride != 0 ? poolFeeOverride : defaultRevenueSharingFeePercentage;
+
+            (uint256 ownerAmount, uint256 treasuryAmount) = _computeAmounts(feeCollectorBptBalance, feePercentage);
+
+            // owner doesn't get the tokens always
+            if (ownerAmount > 0) {
+                // mutate array, and set the owner amount for owner withdrawal
+                amounts[0] = ownerAmount;
+                protocolFeesCollectorMemory.withdrawCollectedFees(tokens, amounts, beneficiary);
+            }
+
+            // mutate array, and set the treasury amount for treasury withdrawal
+            amounts[0] = treasuryAmount;
+            protocolFeesCollectorMemory.withdrawCollectedFees(tokens, amounts, treasury);
+            emit FeesCollected(poolId, beneficiary, ownerAmount, treasury, treasuryAmount);
+        }
+    }
+
+    function _computeAmounts(uint256 feeCollectorBptBalance, uint256 feePercentage)
+        internal
+        pure
+        returns (uint256 ownerAmount, uint256 treasuryAmount)
+    {
+        ownerAmount = Math.divDown(Math.mul(feeCollectorBptBalance, feePercentage), 1e18);
```

Comment:
So as I understand it is possible that balancer will not receive anything from the pool?
If feePercentage would be equal to 100

If yes, is it ok?

Reply:
It is possible that the pool owner doesn't get anything.
If the default fee is 0
If there is no pool override
If the pool owner doesn't register beneficiary for his pool by calling `setPoolBeneficiary`